noir/scripts/sync.sh

```diff
@@ -113,21 +107,10 @@ function has_patch_commit {
   fi
 }
 
-# Find the commit hash of the last applied fixup commit.
-# This is a commit we don't want to include in a patch file.
-function find_fixup_commit {
-  ! repo_exists && return 1
-  fixup=$(git -C noir-repo log --oneline --no-abbrev-commit --grep "$FIXUP_COMMIT_MSG" --max-count=1 | awk '{print $1}')
-  [ -z "$fixup" ] && return 1
-  echo $fixup
-}
-
 # Find the commit hash of the last checkout.
 function find_checkout_commit {
   ! repo_exists && return 1
-  fixup=$(find_fixup_commit)
-  [ -z "$fixup" ] && return 1
-  git -C noir-repo rev-parse "$fixup~1"
+  git -C noir-repo rev-list -n 1 $(read_wanted_ref)
```

Comment:
What does this command do exactly?

Reply:
It looks like if I invoke it with a branch name it gives me the last commit of that branch. What if I'm on a feature branch, like when I was working on `af/msgpack-codegen`, and have the following commits in my history: `origin/af/msgpack-codegen; patch1; patch2; patch-marker; commit1; commit2`. Wouldn't it return `commit2`?

Reply:
From `git rev-list --help`

```
List commits that are reachable by following the parent links from the given commit(s)
```

I'm limiting this command to only return 1 commit hash so it will always return the commit hash which the tag returned from $(read_wanted_ref`)` resolves to.